types.d.ts

```diff
@@ -2532,42 +2416,26 @@ export declare class Connection extends EventEmitter<connectionEventCallback, Co
  * Contains application statistics for a specified time interval and time period.
  */
 export declare class Stats {
-  /**
-   * A {@link StatsMessageTypes} object containing the aggregate count of all message stats.
-   */
-  all: StatsMessageTypes;
-  /**
-   * A {@link StatsRequestCount} object containing a breakdown of API Requests.
-   */
-  apiRequests: StatsRequestCount;
-  /**
-   * A {@link StatsResourceCount} object containing a breakdown of channels.
-   */
-  channels: StatsResourceCount;
-  /**
-   * A {@link StatsConnectionTypes} object containing a breakdown of connection related stats, such as min, mean and peak connections.
-   */
-  connections: StatsConnectionTypes;
-  /**
-   * A {@link StatsMessageTraffic} object containing the aggregate count of inbound message stats.
-   */
-  inbound: StatsMessageTraffic;
   /**
    * The UTC time at which the time period covered begins. If `unit` is set to `minute` this will be in the format `YYYY-mm-dd:HH:MM`, if `hour` it will be `YYYY-mm-dd:HH`, if `day` it will be `YYYY-mm-dd:00` and if `month` it will be `YYYY-mm-01:00`.
    */
   intervalId: string;
   /**
-   * A {@link StatsMessageTraffic} object containing the aggregate count of outbound message stats.
+   * For entries that are still in progress, such as the current month: the last sub-interval included in this entry (in format yyyy-mm-dd:hh:mm:ss), else undefined.
+   */
+  inProgress?: string;
+  /**
+   * The statistics for this time interval and time period. See the JSON schema which the {@link Stats.schema | `schema`} property points to for more information.
    */
-  outbound: StatsMessageTraffic;
+  entries: Record<string, number>;
```

Comment:
did you consider using a `Partial<Record<string, number>>` here? not necessarily saying we should but it's probably more accurate so would be good to make sure we get this right since it would be a breaking change to switch later